### PR TITLE
Fix bug with SqlAzureDacpacDeployment on WIF service connection

### PR DESCRIPTION
**Pull request by LeftTwixWand**
**Task name**: SqlAzureDacpacDeploymentV1

**Description**: Add missing type to Azure RM Connection types, which allows to use WIF.

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** #18992

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by LeftTwixWand** [comment]
@starkmsu The only one thing, which is now clear - is how the Canary tests are finishing fine, when WIF scenario for this task should be broken?

File: Tasks/SqlAzureDacpacDeploymentV1/task.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 1,
-        "Minor": 225,
-        "Patch": 1
+        "Minor": 231,
```
Comment (onetocny) [inline Tasks/SqlAzureDacpacDeploymentV1/task.json:19]: You should bump up the version even in other PowerShell tasks that are referencing VstsAzureRestHelpers_.

File: Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1 [hunk 2]
```diff
@@ -100,11 +100,11 @@ function IsAzureRmConnection {
...
-    if (($connectionType -eq $spnConnection) -or ($connectionType -eq $MsiConnection)) {
-        return $true
-    }
-    else {
-        return $false
+    switch ($connectionType) {
+        $spnConnection { $true }
+        $MsiConnection { $true }
+        $wifConnection { $true }
+        Default { $false }
```
Comment (onetocny) [inline Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1:107]: Could you please add some unit tests for this function?

**Review by onetocny — COMMENTED** [review]
Thanks for the PR. I have left two comments there.

**Comment by onetocny** [comment]
> @starkmsu The only one thing, which is now clear - is how the Canary tests are finishing fine, when WIF scenario for this task should be broken?

Because in CI tests it never gets into [this](https://github.com/microsoft/azure-pipelines-tasks/blob/bb45a3522ea9aab0b410170930158561c31080c5/Tasks/SqlAzureDacpacDeploymentV1/SqlAzureActions.ps1#L386-L390) if clause (response never contains string `sp_set_firewall_rule`, see [CI pipeline log](https://dev.azure.com/canarytest/PipelineTasks/_build/results?buildId=48321&view=logs&j=11a7908b-23d6-5eb5-3940-52e898d761bb&t=37ff9709-13d3-51ff-bacb-61aadbe898ad&l=517)). So that `Get-AgentIPRange` returns empty object and task never creates firewall rule [here](https://github.com/microsoft/azure-pipelines-tasks/blob/bb45a3522ea9aab0b410170930158561c31080c5/Tasks/SqlAzureDacpacDeploymentV1/SqlAzureActions.ps1#L427). To test this scenario we need to somehow configure database to return `sp_set_firewall_rule` in the response [here](https://github.com/microsoft/azure-pipelines-tasks/blob/bb45a3522ea9aab0b410170930158561c31080c5/Tasks/SqlAzureDacpacDeploymentV1/SqlAzureActions.ps1#L352).

File: Tasks/SqlAzureDacpacDeploymentV1/task.json (hunk 1 above)
Comment (LeftTwixWand) [reply Tasks/SqlAzureDacpacDeploymentV1/task.json:19]: Yes, I'm trying to do it, but getting some errors for Node 20 generated tasks.

File: Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1 (hunk 2 above)
Comment (LeftTwixWand) [reply Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1:107]: Do we really need to test the switch statement?

File: Tasks/SqlAzureDacpacDeploymentV1/task.json (hunk 1 above)
Comment (LeftTwixWand) [reply Tasks/SqlAzureDacpacDeploymentV1/task.json:19]: Done

File: Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1 (hunk 2 above)
Comment (LeftTwixWand) [reply Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1:107]: I don't really see the value of this unit test.
I used switch just to clean up a code a bit by simplifying nested if statements.

**Review by starkmsu — APPROVED** [review]

**Pull request merged**